socket/CSocket.cpp

```diff
@@ -176,38 +176,36 @@ int CSocket::bind_socket(int socketid, char* ip)
 int CSocket::destroy_socket(int socketid)
 {
 	if(m_pSocket[socketid] == -1 || !m_pSocketInfo[socketid].success) {
-		logprintf("socket_destroy(): Socket ID %d doesn't exist or hasn't been created yet", socketid);
+		logprintf("socket_destroy(): Socket ID %d doesn't exist or hasn't been created yet.", socketid);
 		return 0;
 	}
-	if(m_pSocket[socketid] != (-1)) {
-		m_pSocketInfo[socketid].active_thread = false;
-		if(!m_pSocketInfo[socketid].is_client) {
-			for(int i = 0;i < m_pSocketInfo[socketid].max_clients;i++) {
-				if(m_pSocketInfo[socketid].connected_clients[i] != INVALID_CLIENT_ID) {
-					if(m_pSocketInfo[socketid].ssl) SSL_free(m_pSocketInfo[socketid].ssl_clients[i]);
-					close_socket(m_pSocketInfo[socketid].connected_clients[i]);
-					m_pSocketInfo[socketid].connected_clients[i] = INVALID_CLIENT_ID;
-				}
+	m_pSocketInfo[socketid].active_thread = false;
+	m_pSocketInfo[socketid].listen = false;
+	m_pSocketInfo[socketid].max_clients = 0;
```

Comment:
Wouldn't setting this to 0 satisfy the `for` (line 186) loop's stopping condition even before its first iteration?